datastore/key.nim

```diff
@@ -0,0 +1,111 @@
+import std/sequtils
+import std/strutils
+
+import pkg/questionable
+import pkg/questionable/results
+
+type
+  Namespace* = object
+    field: ?string
+    value: string
+
+  Key* = object
+    namespaces: seq[Namespace]
+
+const
```

Comment:
maybe, this can be re-defined as operators if it makes sense

Reply:
Sorry, missed this comment earlier. Yes, operators do make sense, at least an operator for building a composite `Key` makes sense.

See `proc /*`:
* https://github.com/status-im/nim-datastore/blob/initial_impl/datastore/key.nim#L244-L245
* https://github.com/status-im/nim-datastore/blob/initial_impl/datastore/key.nim#L253-L254
* https://github.com/status-im/nim-datastore/blob/initial_impl/datastore/key.nim#L274-L275

I have an idea for an operator/s re: `Namespace` (for building namespaces from strings or mixed string/Namespace args) and played with it earlier, but it wasn't clear to me how useful it would be versus, say. aliasing `Namespace.init` to `ns` (e.g. `"a:b".ns`). I decided it's probably best to leave that until the PR is farther along.